pkg/subctl/benchmark/throughput.go

```diff
@@ -0,0 +1,182 @@
+
+	When("a pod sends throughput test to a pod in a remote cluster", func() {
+		It("should be able to execute throughput test", func() {
+			RunThroughputTest(f)
+		})
+	})
+})
+
+func RunThroughputTest(f *framework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+
+	By(fmt.Sprintf("Creating a Nettest Server Pods on %q", clusterBName))
+	netshootPodB := NewNettestSServerPod(f, framework.ClusterB, framework.GatewayNode)
+	pods := framework.KubeClients[framework.ClusterB].CoreV1().Pods(f.Namespace)
+	p1, _ := pods.Get(netshootPodB.Name, metav1.GetOptions{})
+
+	remoteIP := p1.Status.PodIP
+
+	By(fmt.Sprintf("Creating a Nettest Client Pods on %q , connecto to server pod ip %q", clusterAName, remoteIP))
+	nettestPodA := NewNettestClientPod(f, framework.ClusterA, framework.GatewayNode, remoteIP)
+
+	By(fmt.Sprintf("Waiting for the client pod %q to exit, returning what client sent", nettestPodA.Name))
+	AwaitFinish(f, framework.ClusterA, nettestPodA)
+}
+
+func AwaitFinish(f *framework.Framework, cluster framework.ClusterIndex, pod *corev1.Pod) {
+	pods := framework.KubeClients[cluster].CoreV1().Pods(f.Namespace)
+
+	_, terminationErrorMsg, _ := framework.AwaitResultOrError(fmt.Sprintf("await pod %q finished", pod.Name), func() (interface{}, error) {
+		return pods.Get(pod.Name, metav1.GetOptions{})
+	}, func(result interface{}) (bool, string, error) {
+		pod = result.(*corev1.Pod)
+
+		switch pod.Status.Phase {
+		case corev1.PodSucceeded:
+			return true, "", nil
+		case corev1.PodFailed:
+			return true, "", nil
+		default:
+			return false, fmt.Sprintf("Pod status is %v", pod.Status.Phase), nil
+		}
+	})
+
+	finished := pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
+	if finished {
+		terminationMessage := pod.Status.ContainerStatuses[0].State.Terminated.Message
+
+		framework.Logf("Pod %q output:\n%s", pod.Name, terminationMessage)
+		if terminationErrorMsg != "" {
+			framework.Logf("Pod %q error:\n%s", pod.Name, terminationErrorMsg)
+		}
+	}
+}
+
+func AwaitReady(f *framework.Framework, cluster framework.ClusterIndex, pod *corev1.Pod) {
+	pods := framework.KubeClients[cluster].CoreV1().Pods(f.Namespace)
+
+	framework.AwaitUntil("await pod ready", func() (interface{}, error) {
+		return pods.Get(pod.Name, metav1.GetOptions{})
+	}, func(result interface{}) (bool, string, error) {
+		p1 := result.(*corev1.Pod)
+		if p1.Status.Phase != corev1.PodRunning {
+			if p1.Status.Phase != corev1.PodPending {
+				return false, "", fmt.Errorf("unexpected pod phase %v - expected %v or %v", pod.Status.Phase, corev1.PodPending, corev1.PodRunning)
+			}
+			return false, fmt.Sprintf("Pod %q is still pending", p1.Name), nil
+		}
+		return true, "", nil // pod is running
+	})
+}
+
+func NewNettestSServerPod(f *framework.Framework, cluster framework.ClusterIndex, scheduling framework.NetworkPodScheduling) *corev1.Pod {
```

Comment:
@sridhargaddam @mangelajo - does it make sense to add those pods to the framework, to part of network_pods?

Reply:
Would make sense IMO, ping @tpantelis